dash_bio/Circos.py

```diff
@@ -30,14 +30,74 @@ class Circos(Component):
+- tracks (dict; optional): Tracks that specify specific layouts.
 For a complete list of tracks and usage,
-please check the docs."""
+please check the docs. tracks has the following type: list of dicts containing keys 'id', 'data', 'config', 'type', 'tooltipContent', 'color'.
+Those keys have the following types:
+  - id (string; optional): The id of a specific piece of track data.
+  - data (list; required): The data that makes up the track. It can
+be a Json object.
+  - config (dict; optional): The layout of the tracks, where the user
+can configure innerRadius, outterRadius, ticks,
+labels, and more.
+  - type (optional): Specify the type of track this is.
+Please check the docs for a list of tracks you can use,
+and ensure the name is typed in all capitals.
+  - tooltipContent (dict; optional): Specify what data for tooltipContent is
+displayed.
+
+The entry for the "name" key, is any of the keys used in the data loaded into tracks.
+Ex: "tooltipContent": {"name": "block_id"},
+
+To display all data in the dataset use "all" as the entry for the key "name".
+Ex: "tooltipContent": {"name": "all"}
+
+Ex: This will return (source) + ' > ' + (target) + ': ' + (targetEnd)'.
+"tooltipContent": {
+                "source": "block_id",
+                "target": "position",
+                "targetEnd": "value"
+                        },
+Ex: This will return (source)(sourceID) + ' > ' + (target)(targetID) + ': ' (target)(targetEnd)'.
+"tooltipContent": {
+                "source": "source",
+                "sourceID": "id",
+                "target": "target",
+                "targetID": "id",
+                "targetEnd": "end"
+            }. tooltipContent has the following type: a value equal to: PropTypes.string, PropTypes.shape({
+    name: PropTypes.string.isRequired,
+}), PropTypes.shape({
+    source: PropTypes.string.isRequired,
+    sourceID: PropTypes.string,
+    target: PropTypes.string.isRequired,
+    targetEnd: PropTypes.string.isRequired,
+    targetID: PropTypes.string,
+})
+  - color (dict; optional): Specify which dictonary key to grab color values from, in the passed in dataset.
+This can be a string or an object.
+
+If using a string, you can specify hex,
+RGB, and colors from d3 scale chromatic (Ex: RdYlBu).
+
+The key "name" is required for this dictionary,
+where the input for "name" points to some list of
+dictionaries color values.
+
+Ex: "color": {"name": "some key that refers to color in a data set"}. color has the following type: a value equal to: PropTypes.string, PropTypes.shape({
+    name: PropTypes.string.isRequired,
+})"""
     @_explicitize_args
     def __init__(self, enableDownloadSVG=Component.UNDEFINED, enableZoomPan=Component.UNDEFINED, id=Component.UNDEFINED, style=Component.UNDEFINED, eventDatum=Component.UNDEFINED, selectEvent=Component.UNDEFINED, layout=Component.REQUIRED, config=Component.UNDEFINED, size=Component.UNDEFINED, tracks=Component.UNDEFINED, **kwargs):
```

Comment:
Could we regenerate this file in order to introduce line breaks here?

Reply:
We can try modifying the JS file. If that doesn't work, it would have to go somewhere in https://github.com/plotly/dash/blob/dev/dash/development/_py_components_generation.py.

Reply:
Absolutely! I'll resolve this conversation and address the issue separately.

Reply:
Issue lives at https://github.com/plotly/dash/issues/931.